src/jdk.jfr/share/classes/jdk/jfr/internal/EventControl.java

```diff
@@ -80,7 +81,7 @@ record NamedControl(String name, Control control) {
         if (eventType.hasThreshold()) {
             addControl(Threshold.NAME, defineThreshold(eventType));
         }
-        if (eventType.hasStackTrace()) {
+        if (eventType.hasStackTrace() && !eventType.isCPUTimeMethodSampling()) {
```

Comment:
The second clause is not really needed, right? I don't think the CPUTimeMethodSampling event hasStacktrace() is true.

The event has a separate StackTrace field in the event, but that is not what is being checked here.

Reply:
It is needed, because otherwise the `TestDefaultConfigurations` test fails:

```
java.lang.Exception: Setting 'stackTrace' in event 'jdk.CPUTimeSample' was not configured in the configuration 'default'
Setting 'stackTrace' in event 'jdk.CPUTimeSample' was not configured in the configuration 'profile'

        at jdk.jfr.event.metadata.TestDefaultConfigurations.throwExceptionWithErrors(TestDefaultConfigurations.java:115)
        at jdk.jfr.event.metadata.TestDefaultConfigurations.main(TestDefaultConfigurations.java:76)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:573)
        at com.sun.javatest.regtest.agent.MainWrapper$MainTask.run(MainWrapper.java:138)
        at java.base/java.lang.Thread.run(Thread.java:1575)
```

Reply:
The problem is, is that `eventType.hasStackTrace()` checks for the existence of a field called `stackTrace`, which CPUTimeSample events have. We could of course rename the field to something like `recordedStackTrace`, but I don't think that this is helpful.

Reply:
Shouldn't the event have a stackTrace setting and the setting be set to true in default.jfc, similar to ObjectAlllocationSample?
```
    <event name="jdk.ObjectAllocationSample">
      <setting name="enabled">true</setting>
      <setting name="throttle">150/s</setting>
      <setting name="stackTrace">true</setting>
    </event>
```

Reply:
This would also be an option.

Reply:
> The problem is, is that `eventType.hasStackTrace()` checks for the existence of a field called `stackTrace`, which CPUTimeSample events have. We could of course rename the field to something like `recordedStackTrace`, but I don't think that this is helpful.

Aha - so even if the event is not decorated with stackTrace=true in metadata.xml, it has an explicit field with the same name "StackTrace". Wonder why this should be handled differently from ExecutionSample?

Reply:
> Aha - so even if the event is not decorated with stackTrace=true in metadata.xml, it has an explicit field with the same name "StackTrace".

Yes.

> Wonder why this should be handled differently from ExecutionSample?

You're not the only one. I'm looking into it today.

Reply:
Could it be that the ExecutionSample is a periodic event, CPUTimeSample is not.

Reply:
That makes sense. But can I just keep the code that I wrote then?

Reply:
Can't you do the same as ObjectAllocationSample and OldObjectSample? They are not periodic.

Reply:
And they set a `<setting name="stackTrace">true</setting>` in the settings. But this doesn't make sense here, because without a stacktrace, a sample is not really that useful.

Reply:
If you disable stackTrace, you will still get the thread. Perhaps not that useful, but that could be said about many events that have a stack trace. I think we should keep it consistent with other events.

Reply:
Stack trace is enabled by default, so a user should not need to do additional work with a stackTrace setting.

I could be wrong, but I think the reason the ExecutionSample event looks like that is due to legacy reasons. Then we just copied that when we created the NativeSample event. Maybe we should change ExecutionSample and NativeSample so they work like other events (but not in this PR)?

I can look into it in the next few days and see if there is a reason they don't have a stackTrace setting.

Reply:
> Maybe we should change ExecutionSample and NativeMethodSample so they work like other events (but not in this PR)?

Good idea.

Ok, then I change the code accordingly.